CMakeLists.txt

```diff
@@ -115,6 +133,31 @@ add_executable(list_cpu_features src/utils/list_cpu_features.c)
 target_link_libraries(list_cpu_features PRIVATE cpu_features)
 add_executable(CpuFeature::list_cpu_features ALIAS list_cpu_features)
 
+#
+# library : NDK compat
+#
+if(UNIX)
+  find_package(Threads)
+  set (NDK_COMPAT_HDRS ndk_compat/cpu-features.h)
+  set (NDK_COMPAT_SRCS
+    ndk_compat/cpu-features.c
+    $<TARGET_OBJECTS:utils>
+    $<TARGET_OBJECTS:linux_based_hardware_detection>
+  )
+  add_arch_sources(NDK_COMPAT_HDRS NDK_COMPAT_SRCS)
+  add_library(ndk_compat ${NDK_COMPAT_HDRS} ${NDK_COMPAT_SRCS})
+  setup_include_and_definitions(ndk_compat)
+  target_link_libraries(ndk_compat PUBLIC ${CMAKE_DL_LIBS} ${CMAKE_THREAD_LIBS_INIT})
+endif()
+
+#
+# program : NDK compat test program
+#
+if(UNIX)
+  add_executable(ndk-compat-test ndk_compat/ndk-compat-test.c)
+  target_link_libraries(ndk-compat-test PUBLIC ndk_compat)
```

Comment:
I would say `PRIVATE`, but since you usually don't link against an executable...